packages/hash/frontend/src/blocks/blocksMeta.tsx

```diff
@@ -1,23 +1,24 @@
 import { BlockMeta } from "@hashintel/hash-shared/blockMeta";
-import { createContext, useContext, useMemo, useState } from "react";
+import { createContext, useContext, useMemo } from "react";
+import { useLocalstorageState } from "rooks";
 
-type BlocksMetaMap = Map<string, BlockMeta>;
+export type BlocksMetaMap = Record<string, BlockMeta>;
 
 interface BlocksMetaContextState {
   value: BlocksMetaMap;
   setValue(value: BlocksMetaMap): void;
 }
 
 const BlocksMetaContext = createContext<BlocksMetaContextState>({
-  value: new Map(),
+  value: {},
   setValue(_) {},
```

Comment:
I recognise you're not setting this up in this PR, so take this or leave it, but:

It might be worth at some point setting the default for `BlocksMetaContext` to be `null`, and then `useBlocksMeta` can throw if its not set (which is part of why we tend to wrap `useContext`) – otherwise if `BlocksMetaProvider` is accidentally removed then we'll get a silent failure on functionality that depends on it.

Reply:
I went down this route and allowing `null` requires too much type-trickery to for too little gain of robustness. Instead, I now hold on to the reference of the initial empty map and throw an error should `useBlocksMeta` return it.

Reply:
I'm not sure I understand why allowing null requires type trickery. It should be as simple as this.

```ts
type MyContext = { foo: "bar"; }
const MyContext = createContext<MyContext | null>(null);
const useMyContext = (): MyContext => {
   const value = useContext(MyContext);
   if (value === null) throw new Error("You must use useMyContext within a provider");

   return value;
};
```

Unless I'm missing something, that's job done. 

Anyway, this isn't really the purpose of your PR, and sounds like you found a solution that works.

Reply:
Alright, now I get it. I though you meant to allow null only for the `value` property of my `BlocksMetaContextState`. Fixed that, now.